### PR TITLE
adds some debugging info

### PR DESCRIPTION
**Pull request by johnnyman727**

### CODE REVIEW

File: debugging/usb.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Debugging USB Communications
+
+### Introduction to USB
+
+For an in-depth description of how USB works, read [about a code deploy example](https://github.com/tessel/onboarding/blob/master/T2-TECHNICAL-OVERVIEW.md#a-code-deploy-example). Otherwise, the TL;DR is that USB communication from the CLI gets sent to the atmel samd21 coprocessor which then routes the packets over SPI to a daemon running on OpenWRT which then passes it on to another daemon whose only job is to create processes requested over USB and pipe the standard streams back to it. If that description caused your eyes to glaze over, that's okay because the only thing you need to take out of that is that for USB comms to be working, you need:
```
Comment (Frijol) [inline debugging/usb.md]: hmm if you know it will cause eyes to glaze over (mine did by the second sentence) think about rewriting it/breaking it up.

**Comment by Frijol** [comment]
This is wonderfully useful. However, let's disambiguate what should go on tessel/t2-docs vs tessel/onboarding - this is the same type of doc as the SSH and dterm documentation that live there.

The distinction in my mind:
* t2-docs is documentation for people who are using Tessels to develop their own projects
* onboarding is documentation for people who are developing Tessel 2 itself

**Comment by johnnyman727** [comment]
@Frijol agreed on confusion between `t2-docs` and `onboarding`. My issue with `onboarding` is that there are non Team Members with T2s (and there will eventually be lots more) who might want to debug and wouldn't think to look in a repo called `onboarding`

**Comment by Frijol** [comment]
perhaps we should change the name of the onboarding repo? I think it's a useful distinction - particularly, I don't want T2 users thinking they'll need to debug USB or use dterm in normal usage.

File: debugging/lan_discovery.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+This tells the `dns-sd` (DNS Service Discovery) tool to start searching for any devices advertising as a `_tessel._tcp` device. If your device doesn't show up in the results, then it is an issue with detection, then your computer isn't detecting it up properly.
+
+There is also issue a peculiar but uncommon issue where OSX just stops picking up the advertisements for some reason. The cause is unknown thus far but you can fix it with:
+```
+sudo killall -HUP mDNSResponder
+```
+This command basically clears the cache of mDNS devices so the next time you run the `dns-sd` utility or `t2 list`, Tessel will show up (if this was the root cause).
+
+## An issue with network configuration
+There could also be an issue with the network you're computer is using where it blocks mDNS traffic. We've seen the latter case numerous times on company wifi networks. If it is a network configuration issue - you'll have solve that social engineering problem on your own.
```
Comment (Frijol) [inline debugging/lan_discovery.md]: how would you know if this was the problem?

Comment (johnnyman727) [inline debugging/lan_discovery.md]: Good question - I guess if the other debugging steps don't work? Maybe I should put this section at the end?

**Comment by johnnyman727** [comment]
@Frijol I would recommend we either name this file appropriately in the docs repo (like 'developer-debugging') or create a new repo. There are people outside of the contributing team who will want to see it and it's too hard to find in `onboarding`.

**Comment by Frijol** [comment]
What's the story for discoverability?

**Comment by johnnyman727** [comment]
I would just keep it in this repo (not pull it into the t2-start experience). I think this would be a reasonable location and easy to remember.

**Pull request merged**